src/test/java/net/masterthought/cucumber/json/EmbeddingTest.java

```diff
@@ -133,11 +128,13 @@ public void getExtension_ReturnsFileExtension() {
         assertThat(actualExtension).isEqualTo(this.fileName.split("\\.")[1]);
     }
 
-    @Test
-    public void getExtension_UsesExtensionFromNameWhenMIMETypeIsUnknown() {
+    @MethodSource("data")
+    @ParameterizedTest(name = "\"{0}\" with \"{1}\"")
+    public void getExtension_UsesExtensionFromNameWhenMIMETypeIsUnknown(String mimeType, String data, String decodedData, String fileName) {
+        initEmbeddingTest(mimeType, data, decodedData, fileName);
         // Arrange
-        String mimeType = "unknown/mimetype";
-        String data = "c29tZSBkYXRh";
+        mimeType = "unknown/mimetype";
```

Comment:
why did you loose type ?

Reply:
Simple, it was already defined and refused to compile until I did so :)